rotkehlchen/chain/substrate/manager.py

```diff
@@ -0,0 +1,555 @@
+                    f'is not synced with the chain. Node last block is {last_block}, '
+                    f'expected last block is {metadata_last_block}. '
+                    f'Balances and other queries may be incorrect.',
+                )
+
+        return last_block
+
+    def _connect(self) -> None:
+        node_name = self.chain.get_node_name()
+        for node in node_name:
+            self.greenlet_manager.spawn_and_track(
+                after_seconds=None,
+                task_name=f'Substrate {self.chain} manager connection to {node} node',
+                method=self._connect_node,
+                node=node,
+            )
+
+    def _connect_node(self, node: NodeName) -> Tuple[bool, str]:
+        """Attempt to connect to a node, check its status and store its
+        attributes (e.g. interface, weight) in the available nodes map.
+        """
+        if node in self.available_node_attributes_map:
+            msg = f'{self.chain} already connected to {node} node at endpoint: {node.endpoint()}.'
+            return True, msg
+
+        try:
+            node_interface = self._get_node_interface(node)
+            self._check_chain_id(node_interface)
+            last_block = self._check_node_synchronization(node_interface)
+            self._set_chain_properties(node_interface)
+        except RemoteError as e:
+            return False, str(e)
+
+        log.info(f'{self.chain} connected to {node} node at endpoint: {node_interface.url}.')
+        self.available_node_attributes_map[node] = NodeNameAttributes(
+            node_interface=node_interface,
+            weight_block=last_block,
+        )
+        return True, ''
+
+    def _get_account_balance(
+            self,
+            account: SubstrateAddress,
+            node_interface: NodeInterface,
+    ) -> FVal:
+        """Given an account get its amount of chain native token.
+
+        Substrate AccountData documentation:
+        https://substrate.dev/rustdocs/v2.0.0/pallet_balances/struct.AccountData.html
+
+        Polkadot balance documentation:
+        https://wiki.polkadot.network/docs/en/learn-accounts#balance-types
+        """
+        log.debug(
+            f'{self.chain} querying {self.chain_properties.token.identifier} balance',
+            url=node_interface.url,
+            account=account,
+        )
+        try:
+            result = node_interface.query(
+                module='System',
+                storage_function='Account',
+                params=[account],
+            )
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(
+                f'{self.chain} could not connect node at endpoint: {node_interface.url}. '
+                f'Connection error: {str(e)}.',
+            ) from e
+        except (SubstrateRequestException, ValueError) as e:
+            # ValueError: when account is not valid for the given chain, but it
```

Comment:
It is true that the address has been already checked (we check its `ss58_format` for the given chain).

The string contained is `Invalid Address type`, but `ValueError` can be raised by [many other reasons](https://github.com/polkascan/py-substrate-interface/blob/master/substrateinterface/base.py#L1108). If we assume any `ValueError` raised will be because of our implementation and it should only happen during the development, I can remove it.